test/jit/test_generator.py

```diff
@@ -0,0 +1,133 @@
+# Owner(s): ["oncall: jit"]
+
+import io
+import math
+
+import torch
+from torch.nn import init
+from torch.testing._internal.jit_utils import JitTestCase
+from torch.testing._internal.common_utils import skipIfLegacyJitExecutor
+
+
+if __name__ == "__main__":
+    raise RuntimeError(
+        "This test file is not meant to be run directly, use:\n\n"
+        "\tpython test/test_jit.py TESTNAME\n\n"
+        "instead."
+    )
+
+
+class TestGenerator(JitTestCase):
+    @skipIfLegacyJitExecutor("legacy JIT executor does not support Generator type")
+    def test_trace(self):
+        def f():
+            generator = torch.Generator()
+            generator.seed()
+            generator.manual_seed(2023)
+            generator.initial_seed()
+            tensor = torch.empty(2, 2)
+            tensor.uniform_(0, 1, generator=generator)
+            return tensor
+
+        torch.manual_seed(1)
+
+        eager_tensor = f()
+
+        # Change the seed of the default generator to
+        # check that we're using the generator from the
+        # trace
+        torch.manual_seed(2)
+
+        traced_f = torch.jit.trace(f, ())
+        traced_tensor = traced_f()
+
+        self.assertEqual(eager_tensor, traced_tensor)
+
+    def test_script(self):
+        def f():
+            generator = torch.Generator()
+            generator.seed()
+            generator.manual_seed(2023)
+            generator.initial_seed()
+            tensor = torch.empty(2, 2)
+            tensor.normal_(-1.0, 1.0, generator=generator)
+            return tensor
+
+        torch.manual_seed(1)
+
+        eager_tensor = f()
+
+        # Change the seed of the default generator to
+        # check that we're using the generator from the
+        # trace
+        torch.manual_seed(2)
+
+        script_f = torch.jit.script(f, ())
+        script_tensor = script_f()
+
+        self.assertEqual(eager_tensor, script_tensor)
+
+    def test_default_generator(self):
+        def f():
+            # check that calling manual seed for the default generator works
+            torch.manual_seed(2023)
+            tensor = torch.empty(2, 2)
+            tensor.normal_(-1.0, 1.0)
+            return tensor
+
+        torch.manual_seed(1)
+
+        eager_tensor = f()
+
+        torch.manual_seed(2)
+
+        script_f = torch.jit.script(f, ())
+        script_tensor = script_f()
+
+        self.assertEqual(eager_tensor, script_tensor)
+
+    def test_save_load(self):
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.foo = torch.nn.Linear(2, 2)
+                self.bar = torch.nn.Linear(2, 2)
+
+                self.reset_parameters()
```

Comment:
so this doesn't actually test generators in torchscript; the `__init__` method doesn't get scripted, only the `forward`. Could you update with an example that uses a generator in the forward method?

Also I imagine things might be more difficult to save/load if you have a `self.generator = Generator()..` - because then you'll have to be able to serialize the Generator itself. Can you add a test with this as well? If it doesn't work I think that might also be okay, but maybe add some notes somewhere clarifying that